operationsgateway_api/src/routes/ingest_data.py

```diff
@@ -56,16 +57,22 @@ async def submit_hdf(
         record.store_thumbnail(waveform)
 
     log.debug("Processing images")
-    for i in images:
-        image = Image(i)
-        image.store()
+    image_instances = [Image(i) for i in images]
+    for image in image_instances:
         image.create_thumbnail()
         record.store_thumbnail(image)
 
+    # Upload images to Echo S3 using a thread pool to be more efficent when uploading
+    # multiple images from the same HDF file
+    if len(image_instances) > 0:
+        pool = ThreadPool(processes=len(image_instances))
+        pool.map(Image.upload_image, image_instances)
```

Comment:
I think this is quite an important point for production use but it's probably too big to implement into this PR - I'd like to get the Echo changes into `main` in so it can be easily deployed to the dev server before the simulated data is put on there. I have created an issue (DSEGOG - 271) so we keep track of this and can revisit at a later date